aospy/test/test_utils_times.py

```diff
@@ -130,49 +130,118 @@ def test_datetime_or_default(self):
-            pd.Timestamp.min.year + 1
+            pd.Timestamp.min.year + 1, pd.Timestamp.min.year + 2
         ), datetime.datetime(pd.Timestamp.min.year + 1, 1, 1))
 
         self.assertEqual(
-            numpy_datetime_range_workaround(datetime.datetime(3, 1, 1), 1),
+            numpy_datetime_range_workaround(datetime.datetime(3, 1, 1), 1, 6),
             datetime.datetime(pd.Timestamp.min.year + 3, 1, 1)
         )
 
         self.assertEqual(
-            numpy_datetime_range_workaround(datetime.datetime(5, 1, 1), 4),
+            numpy_datetime_range_workaround(datetime.datetime(5, 1, 1), 4, 6),
             datetime.datetime(pd.Timestamp.min.year + 2, 1, 1)
         )
 
-    def test_numpy_datetime_workaround_encode_cf(self):
-        # 255169 days from 0001-01-01 corresponds to date 700-02-04.
-        days = 255169.
-        time = xr.DataArray([days], dims=[TIME_STR])
-        ds = xr.Dataset(coords={TIME_STR: time})
-        ds[TIME_STR].attrs['units'] = 'days since 0001-01-01 00:00:00'
-        ds[TIME_STR].attrs['calendar'] = 'noleap'
-        actual, min_year = numpy_datetime_workaround_encode_cf(ds)
-
-        time_desired = xr.DataArray([days], dims=[TIME_STR])
-        desired = xr.Dataset(coords={TIME_STR: time_desired})
-        desired[TIME_STR].attrs['units'] = (
-            'days since {0}-01-01 00:00:00'.format(979)
+        # Test min_yr outside valid range
+        self.assertEqual(
+            numpy_datetime_range_workaround(
+                datetime.datetime(pd.Timestamp.min.year + 3, 1, 1),
+                pd.Timestamp.min.year, pd.Timestamp.min.year + 2),
+            datetime.datetime(pd.Timestamp.min.year + 4, 1, 1)
         )
-        desired[TIME_STR].attrs['calendar'] = 'noleap'
 
-        assert actual.identical(desired)
+        # Test max_yr outside valid range
+        self.assertEqual(
+            numpy_datetime_range_workaround(
+                datetime.datetime(pd.Timestamp.max.year + 2, 1, 1),
+                pd.Timestamp.max.year - 1, pd.Timestamp.max.year),
+            datetime.datetime(pd.Timestamp.min.year + 4, 1, 1)
+        )
+
+    def test_numpy_datetime_workaround_encode_cf(self):
+        def create_test_data(days, ref_units, expected_units):
+            time = xr.DataArray([days, days + 1095.], dims=[TIME_STR])
+            ds = xr.Dataset(coords={TIME_STR: time})
+            ds[TIME_STR].attrs['units'] = ref_units
+            ds[TIME_STR].attrs['calendar'] = 'noleap'
+            actual, min_yr, max_yr = numpy_datetime_workaround_encode_cf(ds)
+
+            time_desired = xr.DataArray([days, days + 1095.], dims=[TIME_STR])
+            desired = xr.Dataset(coords={TIME_STR: time_desired})
+            desired[TIME_STR].attrs['units'] = expected_units
+            desired[TIME_STR].attrs['calendar'] = 'noleap'
+            return actual, min_yr, max_yr, desired
+
+        # 255169 days from 0001-01-01 corresponds to date 700-02-04.
```

Comment:
Something like this line is what I have in mind in the above comment re: the 1095 value